sklearn/decomposition/tests/test_incremental_pca.py

```diff
@@ -384,3 +384,16 @@ def test_incremental_pca_partial_fit_float_division():
 
     np.testing.assert_allclose(singular_vals_float_samples_seen,
                                singular_vals_int_samples_seen)
+
+
+def test_incremental_pca_fit_overflow_error():
+    rng = np.random.RandomState(0)
+    A = rng.rand(500000, 2)
+
+    ipca = IncrementalPCA(n_components=2, batch_size=10000)
+    ipca.fit(A)
+
+    pca = PCA(n_components=2)
+    pca.fit(A)
+
+    np.testing.assert_allclose(ipca.singular_values_, pca.singular_values_)
```

Comment:
So the issue is that this test passes for me on master, even without this PR. It fails for you?

Reply:
this fails for me without the fix.

```
>>> from sklearn.decomposition import IncrementalPCA, PCA
>>> import numpy as np
>>>
>>> rng = np.random.RandomState(0)
>>> A = rng.rand(500000, 2)
>>>
>>> ipca = IncrementalPCA(n_components=2, batch_size=10000)
>>> ipca.fit(A)
C:\Users\Amanda\Miniconda3\envs\ai_env\lib\site-packages\sklearn\decomposition\_incremental_pca.py:293: RuntimeWarning: overflow encountered in long_scalars
  np.sqrt((self.n_samples_seen_ * n_samples) /
C:\Users\Amanda\Miniconda3\envs\ai_env\lib\site-packages\sklearn\decomposition\_incremental_pca.py:294: RuntimeWarning: invalid value encountered in sqrt
  n_total_samples) * (self.mean_ - col_batch_mean)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Amanda\Miniconda3\envs\ai_env\lib\site-packages\sklearn\decomposition\_incremental_pca.py", line 215, in fit
    self.partial_fit(X_batch, check_input=False)
  File "C:\Users\Amanda\Miniconda3\envs\ai_env\lib\site-packages\sklearn\decomposition\_incremental_pca.py", line 298, in partial_fit
    U, S, V = linalg.svd(X, full_matrices=False)
  File "C:\Users\Amanda\Miniconda3\envs\ai_env\lib\site-packages\scipy\linalg\decomp_svd.py", line 109, in svd
    a1 = _asarray_validated(a, check_finite=check_finite)
  File "C:\Users\Amanda\Miniconda3\envs\ai_env\lib\site-packages\scipy\_lib\_util.py", line 239, in _asarray_validated
    a = toarray(a)
  File "C:\Users\Amanda\Miniconda3\envs\ai_env\lib\site-packages\numpy\lib\function_base.py", line 499, in asarray_chkfinite
    "array must not contain infs or NaNs")
ValueError: array must not contain infs or NaNs
```

Reply:
Right, it's a Windows specific issue (I'm on Linux)  https://github.com/scikit-learn/scikit-learn/issues/17693#issuecomment-648821808. Could you please mention that in the what's new entry.